firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateMonitor.java

```diff
@@ -254,16 +256,43 @@ public void registerForAppState(WeakReference<AppStateCallback> client) {
    */
   /** @hide */
   public void unregisterForAppState(WeakReference<AppStateCallback> client) {
-    synchronized (mClients) {
-      mClients.remove(client);
+    synchronized (appStateClients) {
+      appStateClients.remove(client);
+    }
+  }
+
+  /**
+   * Register a client to receive app cold start update.
+   *
+   * @param client an AppColdStartCallback instance.
+   * @hide
+   */
+  /** @hide */
+  public void registerForAppColdStart(AppColdStartCallback client) {
+    synchronized (appStateClients) {
+      appColdStartClients.add(client);
+    }
+  }
+
+  /**
+   * Unregister the client to stop receiving app cold start update.
+   *
+   * @param client an AppColdStartCallback instance.
+   * @hide
+   */
+  /** @hide */
+  public void unregisterForColdStart(AppColdStartCallback client) {
+    synchronized (appStateClients) {
+      appColdStartClients.remove(client);
```

Comment:
Should we check if the client is already in the list before attempting to remove? What happens if the client is not there - Would it lead it a crash?

Reply:
It won't crash https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#remove(java.lang.Object)

However, I don't think we are unregistering it anywhere? May I am missing it.

Reply:
We are not unregistering it anywhere. I added this method just for completion. Should I remove this method?

Reply:
To answer my own question, dead code is bad "Engineers spend time reading it for no value. They spend additional time trying to puzzle out why it's there, until they convince themselves there is no reason." (go/deadcode). So I will delete this method.